heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -179,6 +189,60 @@ public Boolean deleteSchedulerLocation(String topologyName) {
     return awaitResult(delegate.getSchedulerLocation(null, topologyName));
   }
 
+  /**
+   * Checks to see if the execution state exists
+   *
+   * @return a boolean indicating whether the execution state exists or not
+   */
+  public boolean doesExecutionStateExist(String topologyName) {
+    return awaitResult(delegate.executionStateExists(topologyName));
+  }
+
+  /**
+   * Clean all states of a heron topology
+   * 1. Topology def and ExecutionState are required to exist to delete
+   * 2. TMasterLocation, SchedulerLocation and PhysicalPlan may not exist to delete
+   */
+  public Boolean cleanState(String topologyName) {
```

Comment:
I'm not a fan of util methods for things that do statefull operations on the system. IMO utils should be simple static helpers, like string or date helpers.

It's also good IMO to keep this as an instance method here to keep the logic together. If a new state method gets added to this class, it should be clear that this method also in this class needs to be updated. If cleanState were in some utility that could be lost.